client/driver/rkt.go

```diff
@@ -435,66 +472,181 @@ func (d *RktDriver) Start(ctx *ExecContext, task *structs.Task) (*StartResponse,
+		if len(driverConfig.Volumes) > 0 {
+			if enabled := d.config.ReadBoolDefault(rktVolumesConfigOption, rktVolumesConfigDefault); !enabled {
+				return nil, fmt.Errorf("%s is false; cannot use rkt volumes: %+q", rktVolumesConfigOption, driverConfig.Volumes)
+			}
+			for i, rawvol := range driverConfig.Volumes {
+				parts := strings.Split(rawvol, ":")
+				readOnly := "false"
+				// job spec:
+				//   volumes = ["/host/path:/container/path[:readOnly]"]
+				// the third parameter is optional, mount is read-write by default
+				if len(parts) == 3 {
+					if parts[2] == "readOnly" {
+						d.logger.Printf("[DEBUG] Mounting %s:%s as readOnly", parts[0], parts[1])
+						readOnly = "true"
+					} else {
+						d.logger.Printf("[WARN] Unknown volume parameter '%s' ignored for mount %s", parts[2], parts[0])
+					}
+				} else if len(parts) != 2 {
+					return nil, fmt.Errorf("invalid rkt volume: %q", rawvol)
 				}
-			} else if len(parts) != 2 {
-				return nil, fmt.Errorf("invalid rkt volume: %q", rawvol)
+				volName := fmt.Sprintf("%s-%s-%d", d.DriverContext.allocID, sanitizedName, i)
+				prepareArgs = append(prepareArgs, fmt.Sprintf("--volume=%s,kind=host,source=%s,readOnly=%s", volName, parts[0], readOnly))
+				prepareArgs = append(prepareArgs, fmt.Sprintf("--mount=volume=%s,target=%s", volName, parts[1]))
 			}
-			volName := fmt.Sprintf("%s-%s-%d", d.DriverContext.allocID, sanitizedName, i)
-			prepareArgs = append(prepareArgs, fmt.Sprintf("--volume=%s,kind=host,source=%s,readOnly=%s", volName, parts[0], readOnly))
-			prepareArgs = append(prepareArgs, fmt.Sprintf("--mount=volume=%s,target=%s", volName, parts[1]))
 		}
-	}
 
-	// Inject environment variables
-	for k, v := range ctx.TaskEnv.Map() {
-		prepareArgs = append(prepareArgs, fmt.Sprintf("--set-env=%s=%s", k, v))
-	}
+		// Inject environment variables
+		for k, v := range ctx.TaskEnv.Map() {
+			prepareArgs = append(prepareArgs, fmt.Sprintf("--set-env=%s=%s", k, v))
+		}
 
-	// Image is set here, because the commands that follow apply to it
-	prepareArgs = append(prepareArgs, img)
+		// Image is set here, because the commands that follow apply to it
+		prepareArgs = append(prepareArgs, img)
 
-	// Check if the user has overridden the exec command.
-	if driverConfig.Command != "" {
-		prepareArgs = append(prepareArgs, fmt.Sprintf("--exec=%v", driverConfig.Command))
-	}
+		// Check if the user has overridden the exec command.
+		if driverConfig.Command != "" {
+			prepareArgs = append(prepareArgs, fmt.Sprintf("--exec=%v", driverConfig.Command))
+		}
+
+		// Add memory isolator
+		prepareArgs = append(prepareArgs, fmt.Sprintf("--memory=%vM", int64(task.Resources.MemoryMB)))
 
-	// Add memory isolator
-	prepareArgs = append(prepareArgs, fmt.Sprintf("--memory=%vM", int64(task.Resources.MemoryMB)))
+		// Add CPU isolator
+		prepareArgs = append(prepareArgs, fmt.Sprintf("--cpu=%vm", int64(task.Resources.CPU)))
+
+	} else {
+		allocACName, err := appctypes.NewACName(allocVolName)
```

Comment:
Where does `allocVolName` come from? If this is entirely controlled by program then maybe you could just use `appctypes.MustACName(allocVolName)`? Same for `localVolName` and `secretsVolName`.

Reply:
If this can't be done, then please use better error messages. Something like `fmt.Errorf("[ERR] driver.rkt: failed to convert %s to ACName for pod manifest for /alloc volume: %v", allocVolName, err)` or something.